File: r_scripts/data_extraction_postgresql.R
library(RPostgres)

# set working directory to directory of script
setwd(dirname(rstudioapi::getActiveDocumentContext()$path))
getwd()

# Load database connection details from environment variables
host <- Sys.getenv("DB_HOST")
port <- Sys.getenv("DB_PORT")
db_name <- Sys.getenv("DB_NAME")
username <- Sys.getenv("DB_USER")
password <- Sys.getenv("DB_PASSWORD")

# Check if all credentials are set
if (is.null(host) | is.null(port) | is.null(db_name) | 
    is.null(username) | is.null(password)) {
  stop("Error: Please set all database environment variables!")
}

# Connect to PostgreSQL database
connection <- dbConnect(
  Postgres(), 
  host = host, port = port, dbname = db_name, 
  user = username, password = password
)

# Path to SQL file consisting of query
sql_file_path <- paste0(
  "../sql_scripts/queries/french_trotting_2023_analysis_data_extraction.sql" 
)

# Read SQL query from file. 
sql <- readLines(sql_file_path, warn = FALSE)  
sql <- paste(sql, collapse = "\n")

# Execute the query and store results in a data frame
data <- dbGetQuery(connection, sql)

# Close the connection
dbDisconnect(connection)

# Print a message to confirm data extraction
cat("Data extraction complete! ", nrow(data), " rows retrieved.")

# Save the data frame as a CSV file for further analysis
file_path <- "../data/french_trotting_2023_raw_data.csv"
write.csv(data, file = file_path, row.names = FALSE, fileEncoding = 'utf-8')

File: py_scripts/data_extraction_postgresql.py
import psycopg2
import os


# Set working directory to the directory of the script
script_dir = os.path.dirname(os.path.abspath(__file__))
os.chdir(script_dir)

# Load database connection details from environment variables
host = os.environ.get("DB_HOST")
port = os.environ.get("DB_PORT")
db_name = os.environ.get("DB_NAME")
username = os.environ.get("DB_USER")
password = os.environ.get("DB_PASSWORD")

# Check if all credentials are set
if not all([host, port, db_name, username, password]):
    raise ValueError("Error: Please set all database environment variables!")

# Connect to PostgreSQL database
connection = psycopg2.connect(
    host=host, port=port, database=db_name, user=username, password=password
)


# Path to SQL file consisting of query
sql_file_path = "../sql_scripts/queries/" + \
    "french_trotting_2019_analysis_data_extraction.sql"
print(sql_file_path)

# Read SQL query from file
with open(sql_file_path, "r", encoding="utf-8") as sql_file:
    sql = sql_file.read()
    
# Execute the query and store results in a data frame
cursor = connection.cursor()
cursor.execute(sql)
data = cursor.fetchall()

# Close the connection
connection.close()

